Expired/Views/ProductListView.swift

```diff
@@ -51,8 +61,10 @@ struct ProductListView: View {
         }
     }
 
-    private var filteredProducts: [Product] {
-        return productStore.products
+    var filteredProducts: [Product] {
+        return productStore.products.filter { product in
+            return Product().filterListView(product: product, selectedFilter: selectedFilter)
+        }
     }
```

Comment:
```suggestion
    private var filteredProducts: [Product] {
        switch selectedFilter {
            case .All:
                return productStore.products
            case .Expired, .ExpiringSoon, .Good:
                return productStore.products.filter{ filterProduct($0, selectedFilter) }
        }
    }
    
    private func filterProduct(_ product: Product, _ selectedFilter: ProductFilter) -> Bool {
        switch selectedFilter {
            case .All:
                return true
            case .Expired:
                return product.isExpired
            case .ExpiringSoon:
                return product.isExpiringSoon
            case .Good:
                return product.isGood
        }
    }
```

- Creating a new object with `Product()` inside of each iteration of `products` is avoidable.
- `Product` class doesn't seem to be the proper place to implement the `filter` function as `Product` represents  a single entity while `filter` involves multiple `Product`.
- If we were to extract the `filter` function into other places, `ProductStore` can be a good option as that's where `products` is declared.

Reply:
```swift
private var filteredProducts: [Product] {
    switch selectedFilter {
        case .All:
            return productStore.products
        case .Expired, .ExpiringSoon, .Good:
            return productStore.products.filter{ $0.status.rawValue == selectedFilter.rawValue }
    }
}
```

Actually this was my first implementation. However filtering the `products` by explicitly declared switch cases is more readable and maintainable than comparing the `rawValue`(String).